k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapStoreUriDecoder.java

```diff
@@ -83,7 +83,7 @@ public static ImapStoreSettings decode(String uri) {
             String[] userInfoParts = userinfo.split(":");
 
             if (userinfo.endsWith(":")) {
```

Comment:
(not related to your changes, but wouldn't `userInfoParts.length < 2 || TextUtils.isEmpty(userInfoParts[1])` be a more reliable check here?)